core/test-utils/src/e2e/ports.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::net::{TcpListener, TcpStream};
+
+/// Return an ephemeral, available port. On unix systems, the port returned will be in the
+/// TIME_WAIT state ensuring that the OS won't hand out this port for some grace period.
+/// Callers should be able to bind to this port given they use SO_REUSEADDR.
+pub fn get_available_port(host: &str) -> u16 {
```

Comment:
This is how narwhal/sui gets random available ports for testing: https://github.com/MystenLabs/sui/blob/bbfaafc17652d221651e835c908028c440f039d7/narwhal/config/src/utils.rs